### PR TITLE
fix: ticket cancellation by user

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5942

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Please outline the implementation compared to other platforms, e.g. Eventbrite and add screenshots.

**Comment by maze-runnar** [comment]
I have implemented is same as eventbrite in the my tickets section
Screenshot from 2020-12-23 10-50-03

**Comment by mariobehling** [comment]
Please add a screenshot of our website to show the change. Every PR should include the screenshots of the changes.

**Comment by mariobehling** [comment]
Please change "Are you sure you would like to cancel this Ticket?" -> "Are you sure you would like to cancel this ticket?"

**Comment by maze-runnar** [comment]
Screenshot from 2020-12-23 16-37-51
Screenshot from 2020-12-23 16-37-46

**Comment by mariobehling** [comment]
So, we need to think about how we want this area to look like, but for now please change the following:
* Rename "Cancel" to "Cancel Ticket"
* Move button to the right of "Not Checked in" and make it into the same size.

**Comment by maze-runnar** [comment]
> Move button to the right of "Not Checked in" and make it into the same size.

they are not buttons and whole upper card is inside <a> tag.

**Comment by maze-runnar** [comment]
Screenshot from 2020-12-23 17-45-05

File: app/components/order-card.js [hunk 1]
```diff
@@ -16,4 +16,23 @@ export default class OrderCard extends Component {
...
     return checkedInUser.length !== 0;
   }
+
+  @action
+  cancelOrder(order_id) {
+    this.set('isLoading', true);
+    const order = this.store.peekRecord('order', order_id, { backgroundReload: false });
+    order.set('status', 'cancelled');
+    order.save()
+      .then(() => {
```
Comment (iamareebjamal) [inline app/components/order-card.js:26]: Please use async await

**Comment by iamareebjamal** [comment]
Still using promises

File: app/components/order-card.js [hunk 2]
```diff
@@ -22,17 +22,15 @@ export default class OrderCard extends Component {
...
-      })
-      .catch(e => {
-        console.error('Error while cancelling order', e);
-        this.notify.error(this.l10n.t('An unexpected error has occurred.'));
-      })
-      .finally(() => {
-        this.set('isLoading', false);
-      });
+    try {
+      order.save();
```
Comment (iamareebjamal) [inline app/components/order-card.js:26]: await

**Comment by mariobehling** [comment]
Please also change the following:
* Delete "Completed", but keep expired or other visible status. No need to have "completed" as a button as it is the same as "Not Checked In" (Once a completed "ticket" has checked in the ticket becomes an "attendee")
* Move any status and "Not checked in" below ticket info, but in box
* Make "Not checked in" into light gray button
* Make "Cancel Ticket" gray (We do not want it stand out too much.

102995359-dd6f0580-4546-11eb-86bd-2b437e1d82ba

For anything that you cannot solve in this issue please open a new issue.

**Comment by maze-runnar** [comment]
Screenshot from 2020-12-24 12-34-21
Screenshot from 2020-12-24 12-30-41

**Comment by maze-runnar** [comment]
Screenshot from 2020-12-24 12-39-12

**Comment by iamareebjamal** [comment]
The free order line has no left margin, please align it correctly as it is just a small change

**Comment by maze-runnar** [comment]
Screenshot from 2020-12-24 15-19-26

**Pull request merged**